src/ScottPlot/Plottable/DataStructures/Segmentedtree.cs

```diff
@@ -53,14 +52,20 @@ public SegmentedTree(T[] data)
 
         public async Task SetSourceAsync(T[] data)
         {
-            if (data == null)
-                throw new Exception("Data cannot be null");
-            sourceArray = data;
-            await Task.Run(() => UpdateTreesInBackground());
+            sourceArray = data ?? throw new ArgumentNullException("Data cannot be null");
+            await Task.Run(() => UpdateTrees());
```

Comment:
This should be fixed in different PR
With old solution we're not awaiting real operation, only creation of task, that will be scheduled
As the result we might begin to work with data, which is not yet updated properly